src/java.base/share/classes/sun/security/ssl/SSLConfiguration.java

```diff
@@ -261,6 +259,11 @@ void setSSLParameters(SSLParameters params) {
             this.applicationProtocols = sa;
         }   // otherwise, use the default values
 
+        String[] ss = params.getSignatureSchemes();
```

Comment:
If we don’t use the cloning getter here (and use the pre-parsed list) it would be more efficient.

Reply:
Yes.  Array copy is a concern of mine, too.  Hopefully, the frozen array feature could help address the array copy issues in the future.

Reply:
Hmm.. I guess the different packages make it really hard to have an internal optimized getter. What about a single concatenated string, it would be immutable „sig1,sig2“..

Reply:
It's an interesting idea.  I would like to choose simple API, in an affordable performance impact.  I don't worry a lot if considering the frozen array enhancement.